ovs-p4rt/sidecar/newovsp4rt.cc

```diff
@@ -2226,113 +2155,83 @@ absl::Status ConfigTunnelTermTableEntry(ovsp4rt::OvsP4rtSession* session,
   if (!status.ok()) {
     LogFailure(insert_entry, detail.getLogTableName());
   }
   return status;
 }
 
-#endif  // ES2K_TARGET
-
-}  // namespace ovsp4rt
-
 //----------------------------------------------------------------------
-// ovsp4rt_str_to_tunnel_type (common)
+// C++ functions that implement the public API.
 //----------------------------------------------------------------------
-enum ovs_tunnel_type ovsp4rt_str_to_tunnel_type(const char* tnl_type) {
-  if (tnl_type) {
-    if (strcmp(tnl_type, "vxlan") == 0) {
-      return OVS_TUNNEL_VXLAN;
-    } else if (strcmp(tnl_type, "geneve") == 0) {
-      return OVS_TUNNEL_GENEVE;
-    }
-  }
-  return OVS_TUNNEL_UNKNOWN;
-}
-
-#if defined(ES2K_TARGET)
 
 //----------------------------------------------------------------------
-// ovsp4rt_config_fdb_entry (ES2K)
+// ConfigFdbEntry (ES2K)
 //
-// learn_info is passed by value because this function makes local
+// learn_info is passed by value because this function may make local
 // modifications to it.
 //----------------------------------------------------------------------
-void ovsp4rt_config_fdb_entry(struct mac_learning_info learn_info,
-                              bool insert_entry, const char* grpc_addr) {
-  using namespace ovsp4rt;
+void ConfigFdbEntry(ClientInterface& client,
+                    struct mac_learning_info learn_info, bool insert_entry,
+                    const char* grpc_addr) {
+  absl::Status status;
 
   // Start a new client session.
-  auto status_or_session = ovsp4rt::OvsP4rtSession::Create(
-      grpc_addr, GenerateClientCredentials(), absl::GetFlag(FLAGS_device_id),
-      absl::GetFlag(FLAGS_role_name));
-  if (!status_or_session.ok()) return;
-
-  // Unwrap the session from the StatusOr object.
-  std::unique_ptr<ovsp4rt::OvsP4rtSession> session =
-      std::move(status_or_session).value();
+  status = client.connect(grpc_addr);
+  if (!status.ok()) return;
+
+  // Fetch P4Info object from server.
   ::p4::config::v1::P4Info p4info;
-  ::absl::Status status =
-      ovsp4rt::GetForwardingPipelineConfig(session.get(), &p4info);
+  status = client.getPipelineConfig(&p4info);
   if (!status.ok()) return;
 
-  /* Hack: When we delete an FDB entry based on current logic  we will not know
-   * we will not know if it's a Tunnel learn FDB or regular VSI learn FDB.
-   * This hack, during delete case check if entry is present in l2_to_tunnel_v4
-   * and l2_to_tunnel_v6. if any of these 2 tables is true then go ahead and
-   * delete the entry.
+  /* In the delete case, we do not know whether this is a Tunnel learn
+   * entry or a regular VSI learn entry. Check for a match in one of
+   * the L2 Tunnel tables and set the appropriate properties in the
+   * learn_info structure.
    */
-
   if (!insert_entry) {
     auto status_or_read_response =
-        GetL2ToTunnelV4TableEntry(session.get(), learn_info, p4info);
+        GetL2ToTunnelV4TableEntry(client, learn_info, p4info);
```

Comment:
PR #728* defines predicate functions that allow us to simplify this code to:
```c
if (HaveL2ToTunnelV4TableEntry(client, learn_info, p4info)) {
  learn_info.is_tunnel = true;
}
```
See commit https://github.com/ipdk-io/networking-recipe/pull/728/commits/175a9f461de27617522469c7286bb405f0763fdc for details.

\* This PR will need to be revised for `newovsp4rt.cc`.